docs/intro-to-dune/index.md

```diff
@@ -0,0 +1,269 @@
+	npx webpack serve --open --mode development --entry ./_build/default/src/$(app)/output/src/$(app)/Index.js
+```
+
+We basically copied the `serve` npm command from `package.json` to create the
+recipe for this rule, but instead of a fixed value for the `--entry` option, we
+use
+
+```
+./_build/default/src/$(app)/output/src/$(app)/Index.js
+```
+
+This means that the entry script served by webpack dev server depends on the
+`app` environment variable, which is provided by `src/counter/Makefile`.
+
+You're now ready to run the new Counter app you created! Go into the
+`src/counter` directory and run
+
+```
+make serve
+```
+
+It should open a new tab in your default browser to <a
+href="http://localhost:8080/" target="_blank" rel="noreferrer
+noopener">http://localhost:8080/</a> and display your Counter component.
+
+:::info
+Feel to delete the `src/Index.re` and `src/dune` files. You won't need them
+anymore for this or later chapters. You can also delete the `scripts` section of
+your `package.json` file.
+:::
+
+Huzzah! You created a new `dune` file to build an app and a `Makefile` to
+serve that app locally. In future chapters, we'll assume that you use the same
+directory structure for each new app you build.
+
+## Exercises
+
+<b>1.</b> Repeat the steps we did for Counter and create a separate app for
+Celsius Converter.
+
+<b>2.</b> Delete `reason-react-ppx` from `src/counter/dune`'s `melange.emit`
+stanza. What compiler errors do you get?
+
+<b>3.</b> You might have noticed that every recipe in the Makefile is prefixed
+by a tab. What happens if you replace the tab in front of the `serve` rule's
+recipe with four spaces?
+
+## Overview
+
+- Dune is the build system we use to build Melange projects
+- The `dune-project` file describes the metadata for your project, primarily:
+  - The version of Dune you're using
+  - The version of the Melange plugin for Dune you're using
+  - The name of your project
+- `dune` files describe things of interest to Dune, for example:
+  - Which directories to include and which to exclude
+  - Which directories contain code that should be transpiled to JavaScript,
+    using the `melange.emit` stanza
+- Make is a build automation tool heavily used in the OCaml world
+  - `Makefile`s contain rules which can run commands based on the target you
+    pass to the `make` command
+
+## Solutions
+
+<b>1.</b> Creating a separate app for Celsius Converter, with its own `dune`,
+`Makefile`, and `Index.re` files, should look something like
+[this](https://github.com/melange-re/melange-for-react-devs/tree/develop/src/celsius-converter-option).
+
+<b>2.</b> Deleting `reason-react-ppx` from `src/counter/dune` will result in a
+compilation error:
+
+```
+File "src/counter/Counter.re", line 5, characters 2-6:
+5 |   <div
+      ^^^^
+Error: Unbound value div
+```
+
+That's because putting `reason-react-ppx` in the `preprocess/pps` field will
+transform function calls to `div` (which isn't defined anywhere) into calls to
```

Comment:
Maybe we can link somewhere around here to the JSX section of the Reason docs, which explain how `<div/>` is converted: https://reasonml.github.io/docs/en/jsx.

Reply:
Added footnote with two links to more JSX info in https://github.com/melange-re/melange-for-react-devs/pull/6/commits/0a6687827b600dd2a8dfa1c7eab0abb0c14a2199